operator/pkg/manifest/shared.go

```diff
@@ -517,6 +525,84 @@ func overlaySetFlagValues(iopYAML string, setFlags []string) (string, error) {
 	return string(out), nil
 }
 
+var defaultSetFlagConvertPaths = []string{
+	"meshConfig.defaultConfig.proxyMetadata",
+}
+
+// convertDefaultIOPMapValues converts default map[string]string values into string.
+func convertDefaultIOPMapValues(outYAML string, setFlags []string) (string, error) {
+	return convertIOPMapValues(outYAML, setFlags, defaultSetFlagConvertPaths)
+}
+
+// convertIOPMapValues converts certain paths of map[string]string values into string.
+func convertIOPMapValues(outYAML string, setFlags []string, convertPaths []string) (string, error) {
+	for _, setFlagConvertPath := range convertPaths {
+		if containParentPath(setFlags, setFlagConvertPath) {
+			var (
+				converter              = map[string]interface{}{}
+				convertedProxyMetadata = map[string]string{}
+				subPaths               = strings.Split(setFlagConvertPath, ".")
+			)
+
+			if err := yaml.Unmarshal([]byte(outYAML), &converter); err != nil {
+				return outYAML, err
+			}
+			originMap, ok := converter["spec"].(map[string]any)
+			if !ok {
+				return outYAML, nil
+			}
+
+			for index, subPath := range subPaths {
```

Comment:
@howardjohn this logic feels rather boilerplate ... are you aware of any existing utilities that we could leverage here?